versioned_docs/version-0.9.18/data.mdx

```diff
@@ -0,0 +1,277 @@
+      ],
+      "exonIntervalFields": [
+          {"index": 12, "name": "start"},
+          {"index": 13, "name": "end"}
+      ]
+    },
+    ... // other configurations of this track
+  }]
+}
+```
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='BEDDBData'/>
+
+
+
+## Data Transform
+
+<span>
+    Gosling supports a diverse set of data transforms, including 
+    {GoslingSchema["definitions"]['DataTransform'].anyOf.map((d, i)=>{
+      let name = d.$ref.replace('#/definitions/', '')
+      /** 
+        seperate camel-case names. 
+        e.g., "JSONParseTransform" => "JSON Parse Transform"  
+      */
+      name = name.replace(/([a-z])([A-Z])/g, '$1 $2').replace(/([A-Z])([A-Z][a-z])/g, '$1 $2')
+        // replace white space ' ' to '-' to avoid the %20 in url
+        return <span key={name}> <a href={`#${name.replace(/ /g, '-').toLowerCase()}`}>
+            {name}
+        </a> {i==GoslingSchema["definitions"]['DataTransform'].anyOf.length-1 ? '. ': ', '} </span>})
+    }
+</span>
+
+
+```javascript
+{
+  "tracks":[{
+    "data": ...,
+    // a list of data transforms can be applied to the data
+    "dataTransform": [
+          { "type": "filter", "field": "type", "oneOf": ["gene"] },
+          { "type": "filter", "field": "strand", "oneOf": ["+"], "not": true }
+    ],
+    "mark": "rect",
+    ...,
+  }]
+}
+```
+
+### Filter Transform
+
+Users can apply three types of filters: `oneOf`, `inRange`, `include`.
+Each filter transform has the following properties:
+
+**Properties of One Of Filter**
+<TableWrapper GoslingSchema={GoslingSchema} objName='OneOfFilter' includeDescription={true}/>
+
+**Properties of In Range Filter**
+<TableWrapper GoslingSchema={GoslingSchema} objName='RangeFilter' includeDescription={true}/>
+
+**Properties of Include Filter**
+<TableWrapper GoslingSchema={GoslingSchema} objName='IncludeFilter' includeDescription={true}/>
+
+### Str Concat Transform
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='StrConcatTransform' includeDescription={true}/>
+
+### Str Replace Transform 
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='StrReplaceTransform' includeDescription={true}/>
+
+### Log Transform
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='LogTransform' includeDescription={true}/>
+
+### Displace Transform 
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='DisplaceTransform' includeDescription={true}/>
+
+### Exon Split Transform 
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='ExonSplitTransform' includeDescription={true}/>
+
+### Coverage Transform
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='CoverageTransform' includeDescription={true}/>
+
+### JSON Parse Transform
+
+<TableWrapper GoslingSchema={GoslingSchema} objName='JSONParseTransform' includeDescription={true}/>
+
+Apart from these data transforms, users can also aggregate data values (min, max, bin, mean, and count). [Read more about data aggregation](#x)
+
```

Comment:
Could we add two missing data transforms?:

[GenomicLengthTransform](https://github.com/gosling-lang/gosling.js/blob/dec0e6434c39c233cf7cc35fe96c21747f37fa95/src/core/gosling.schema.ts#L1170)

[SvTypeTransform](https://github.com/gosling-lang/gosling.js/blob/dec0e6434c39c233cf7cc35fe96c21747f37fa95/src/core/gosling.schema.ts#L1186)

Also, I can open GH issues on this repository whenever I add new data transform to Gosling.js in the future. I think we should have better guidelines regarding this on https://github.com/gosling-lang/gosling.js/blob/master/CONTRIBUTING.md.

Reply:
Just to clarify, these are included in 0.9.18.